troposphere/awslambda.py

```diff
@@ -38,6 +38,10 @@ def validate(self):
             raise ValueError(
                 "You can't specify both 'S3ObjectVersion' and 'ZipFile'"
             )
+        if zip_file and len(zip_file) > 4096:
```

Comment:
I think running on strings only is a good solution for this problem. I like the iteration over the join to see if they are strings too.